src/gpu/nvidia/cudnn_batch_normalization_executor.hpp

```diff
@@ -40,155 +41,159 @@ struct bnorm_exec_base_t {
+                        sc, ih, cuda_stream, arg_scale_bias, bnorm_impl->C());
+
+            auto *x = arg_src.get_native_pointer(ih);
+            auto *dy = arg_diff_dst.get_native_pointer(ih);
+            auto *dx = arg_diff_src.get_native_pointer(ih);
+
+            auto *scale = static_cast<uint8_t *>(
+                    arg_scale_bias.get_native_pointer(ih));
+            auto *bias = scale + bnorm_impl->C() * sizeof(float);
+
+            auto *diff_scale = static_cast<uint8_t *>(
+                    arg_diff_scale_bias.get_native_pointer(ih));
+            auto *diff_bias = diff_scale + (bnorm_impl->C() * sizeof(float));
+
+            auto *save_mean = static_cast<uint8_t *>(
+                    arg_wkspace.get_native_pointer(ih));
+            auto *save_var = save_mean + bnorm_impl->mean_var_size_bytes();
+            auto *wkspace = save_var + bnorm_impl->mean_var_size_bytes();
+            auto *relu_dy = bnorm_impl->fuse_norm_relu()
+                    ? temp_relu_output->get_native_pointer(ih)
                     : nullptr;
 
             std::shared_ptr<bnorm_args_t> args(
                     new bnorm_bwd_args_t(x, dx, dy, save_mean, save_var, scale,
                             bias, diff_scale, diff_bias, wkspace, relu_dy));
-
             bnorm_impl->execute(handle, args);
         });
     }
 
-    template <typename T>
-    void maybe_init_ss(
-            nvidia::sycl_cuda_stream_t *cuda_stream, T, T, bool) const {}
-
-    template <typename T>
-    void maybe_init_ss(nvidia::sycl_cuda_stream_t *cuda_stream,
-            ::sycl::accessor<T, 1, ::sycl::access::mode::write> scale_acc,
-            ::sycl::accessor<T, 1, ::sycl::access::mode::write> bias_acc,
-            bool init_ss) const {
-        if (init_ss) {
-            constexpr T scale_val = 1, bias_val = 0;
-            cuda_stream->interop_task([&](::sycl::handler &cgh) {
-                cgh.fill(scale_acc, scale_val);
-            });
-
-            cuda_stream->interop_task([&](::sycl::handler &cgh) {
-                cgh.fill(bias_acc, bias_val);
-            });
-        }
+    template <typename T = float>
+    void init_scaleshift(cuda_sycl_scoped_context_handler_t &sc,
+            const compat::interop_handle &ih,
+            nvidia::sycl_cuda_stream_t *cuda_stream, T, size_t) const {}
+
+    template <typename T = float>
+    void init_scaleshift(cuda_sycl_scoped_context_handler_t &sc,
+            const compat::interop_handle &ih,
+            nvidia::sycl_cuda_stream_t *cuda_stream,
+            sycl_memory_arg_t<::sycl::access::mode::write> arg_scale_bias,
+            const size_t n) const {
+        T scale_val = 1, bias_val = 0;
+        cuda_stream->interop_task([&](::sycl::handler &cgh) {
+            T *scale_ptr
+                    = static_cast<T *>(arg_scale_bias.get_native_pointer(ih));
+            T *bias_ptr = scale_ptr + n;
+            cuMemsetD32Async(reinterpret_cast<CUdeviceptr>(scale_ptr),
```

Comment:
Does it make sense to check return code?

Reply:
Change made